R/dsmodel.R

```diff
@@ -313,32 +313,33 @@ dsmodel <- function(fun, title="", display = TRUE) {
 		  res <- unique(c(basin$colMatrix))
 		  (length(res) == 1) && !(is.element(0,res))
 		},
-		find.period= function(self, x, y=NULL, initIters=1000, maxPeriod=128, numTries=1,
+		find.period= function(self, x, y=NULL, ..., initIters=1000, maxPeriod=128, numTries=1,
                           epsilon=sqrt(sqrt(.Machine$double.eps)), rangeMult=0){
-		  if(!(!is.null(y) && length(x)==1 && length(y)==1)){
-		    if(is.dspoint(x)){
-		      y=x$y
-		      x=x$x
-		    }
-		    else if(is.vector(x) && length(x)==2){
-		      y=x[[2]]
-		      x=x[[1]]
-		    }
-		    else {
-	  	    stop("dsmodel: expected input formats for find.period's starting point are two scalars, a vector of length two or a dspoint")
-		    }
-		  }
+		  #i dont think this works with ...
+		  #if(!(!is.null(y) && length(x)==1 && length(y)==1)){
+		  #  if(is.dspoint(x)){
+		  #    y=x$y
+		  #    x=x$x
+		  #  }
+		  #  else if(is.vector(x) && length(x)==2){
+		  #    y=x[[2]]
+		  #    x=x[[1]]
+		  #  }
+		  #  else {
+	  	#    stop("dsmodel: expected input formats for find.period's starting point are two scalars, a vector of length two or a dspoint")
+		  #  }
+		  #}
 		  if(!(rangeMult==0 || rangeMult==Inf ||is.null(rangeMult)) && is.null(self$range)){
 		    stop("is.stable with rangeMult!=0 requires range() to have been composed with the model.")
 		  }
 		  #moves all the points untill they are either all infinite, fixed, or outside of range*rangeMult
 		  for(i in 1:numTries) {
-		    startPoint <- self$apply(x,y,iters=initIters,accumulate=FALSE,crop=FALSE)
-		    if(!self$has.diverged(startPoint$x,startPoint$y,rangeMult)){
+		    startPoint <- self$apply(x,y,...,iters=initIters,accumulate=FALSE,crop=FALSE)
+		    if(!self$has.diverged(startPoint[[1]],startPoint[[2]],rangeMult)){

```

Comment:
The inconsistent of [[1]] vs $x is worrisome. What if $x and $y are valid, but in opposite order?

Reply:
Perhaps we should define functions getX and getY? it would refactor a lot of this.

Reply:
Make an issue for [[1]] vs $x, quoting the above concern.  Get rid of the rd files. Then re-push and I'll accept the pull request.